app/models/concerns/waste_carriers_engine/can_use_new_registration_workflow.rb

```diff
@@ -133,21 +138,25 @@ module CanUseNewRegistrationWorkflow
                       to: :construction_demolition_form
 
           transitions from: :waste_types_form,
-                      to: :company_name_form,
+                      to: :your_tier_form,
                       if: :switch_to_lower_tier_based_on_smart_answers?,
                       after: :switch_to_lower_tier
 
           transitions from: :waste_types_form,
-                      to: :cbd_type_form,
+                      to: :your_tier_form,
```

Comment:
Do we have two `from: :waste_types_form, to: :your_tier_form` transitions just because they have different `after`s? I feel like if they have the same journey then we should be setting the tier somewhere outside of the state machine, since it's not really anything to do with it. Then we can have just one transition.

Reply:
yes, is because of the different `:afters`. Currently, I feel like having the state machine handling this is the best choice, rather than having this responsibility going around in a trillion of forms or controller. I believe is more manageable and improves the maintainability. In any case, I feel like currently is too big of a refactoring/task to change the responsibility of setting the tier from the state machine to a controller, hence if that's the way we want going forward let's open a task somewhere and deal with it separately.